src/Stratis.Bitcoin.Features.BlockStore/BlockStoreSignaled.cs

```diff
@@ -48,22 +55,27 @@ public class BlockStoreSignaled : SignalObserver<Block>
             ChainState chainState,
             IConnectionManager connection,
             INodeLifetime nodeLifetime,
-            IAsyncLoopFactory asyncLoopFactory,
             ILoggerFactory loggerFactory,
-            IBlockStoreCache blockStoreCache,
-            string name = "BlockStore")
+            IBlockStoreCache blockStoreCache)
         {
-            this.asyncLoopFactory = asyncLoopFactory;
-            this.blocksToAnnounce = new ConcurrentQueue<ChainedBlock>();
             this.blockStoreLoop = blockStoreLoop;
             this.chain = chain;
             this.chainState = chainState;
             this.connection = connection;
-            this.name = name;
             this.nodeLifetime = nodeLifetime;
             this.logger = loggerFactory.CreateLogger(this.GetType().FullName);
             this.storeSettings = storeSettings;
             this.blockStoreCache = blockStoreCache;
+            this.asyncLock = new AsyncLock();
+
+            // Set interval between batches.
+            this.batchInterval = TimeSpans.Second;
+
+            this.localBatch = new List<ChainedBlock>();
+
+            // Configure batch timer.
+            this.batchTimer = new Timer(this.batchInterval.TotalMilliseconds);
```

Comment:
AsyncLoop is sort of a time, why not use the async loop is there a beneift of a Timer?
Under the hoods the async loops calls Task.Delay which triggers a timer internally when we use async loop we have more control of the code (exception, logging etc..)

Reply:
I think because we need to stop it sometimes we will cause more resources consumption by recreating cancellation tokens and creating new tasks for the timers. Btw Task.Delay uses Timer internally.
Second reason is because from the logic perspective it's not a loop if we do only 1 iteration so I didn't want to add one more misuse of async loop here. 

But we're talking about really small resources consumption differences here so I'm fine with changing timer to async loop if you'd like me to.

Reply:
Indeed for consistency I would use the AsyncLoop.
Unless we decide we want to move all our background loops to timers (which is also a possibility).

Reply:
using async loop here does not make sense, please dont do it

Reply:
ok, I'll just wait for you and Dan to reach consensus on this one. I don't really mind going with any of those 2 approaches but generally I don't like the idea of using smth called loop for a 1 iteration job.

Reply:
using async loop here would be like using it instead of Task.Delay everywhere in the code and saying that async loop does internally use Task.Delay, so it is better